File: Sources/BeagleUI/Sources/Layout/Padding.swift
//
//  Padding.swift
//  BeagleUI
//
//  Created by Daniel Tes on 12/09/19.
//  Copyright © 2019 Daniel Tes. All rights reserved.
//

public struct Padding: Widget {
    let value: PaddingValue
    let child: Widget
}

public struct PaddingValue {
    let top: UnitValue?
    let left: UnitValue?
    let right: UnitValue?
    let bottom: UnitValue?
}

File: src/main/kotlin/br/com/zup/beagleui/framework/widget/layout/Padding.kt
package br.com.zup.beagleui.framework.widget.layout

import br.com.zup.beagleui.framework.widget.core.Widget
import br.com.zup.beagleui.framework.widget.core.UnitValue

data class PaddingValue(
    val top: UnitValue? = null,
    val left: UnitValue? = null,
    val right: UnitValue? = null,
    val bottom: UnitValue? = null
)

data class Padding(
    val value: PaddingValue,
    val child: Widget
) : Widget
